bees/mumblebee/mumblebeefactory.go

```diff
@@ -0,0 +1,203 @@
+/*
+ *    Copyright (C) 2014-2017 Christian Muehlhaeuser
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU Affero General Public License as published
+ *    by the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Authors:
+ *      Martin Schlierf <go@koma666.de>
+ */
+
+package mumblebee
+
+import (
+	"github.com/muesli/beehive/bees"
+)
+
+// MumbleBeeFactory is a factory for MumbleBees.
+type MumbleBeeFactory struct {
+	bees.BeeFactory
+}
+
+// New returns a new Bee instance configured with the supplied options.
+func (factory *MumbleBeeFactory) New(name, description string, options bees.BeeOptions) bees.BeeInterface {
+	bee := MumbleBee{
+		Bee: bees.NewBee(name, factory.ID(), description, options),
+	}
+	bee.ReloadOptions(options)
+
+	return &bee
+}
+
+// ID returns the ID of this Bee.
+func (factory *MumbleBeeFactory) ID() string {
+	return "mumblebee"
+}
+
+// Name returns the name of this Bee.
+func (factory *MumbleBeeFactory) Name() string {
+	return "Mumble"
+}
+
+// Description returns the description of this Bee.
+func (factory *MumbleBeeFactory) Description() string {
+	return "Connects to Mumble"
+}
+
+// Image returns the filename of an image for this Bee.
+func (factory *MumbleBeeFactory) Image() string {
+	return factory.ID() + ".png"
+}
+
+// LogoColor returns the preferred logo background color (used by the admin interface).
+func (factory *MumbleBeeFactory) LogoColor() string {
+	return "#f42929"
+}
+
+// Options returns the options available to configure this Bee.
+func (factory *MumbleBeeFactory) Options() []bees.BeeOptionDescriptor {
+	opts := []bees.BeeOptionDescriptor{
+		{
+			Name:        "address",
+			Description: "Hostname of Mumble server, eg: de.clanwarz.com:64738",
+			Type:        "string",
+			Mandatory:   true,
+		},
+		{
+			Name:        "channel",
```

Comment:
Can you join multiple channels concurrently on a Mumble server? In this case, we could make it ```channels``` of ```[]string``` type. We could also get rid of the ```move``` action then.